src/Processors/Formats/IOutputFormatWithUTF8Validation.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <Processors/Formats/IOutputFormat.h>
+#include <Processors/Formats/IRowOutputFormat.h>
+
+#include <IO/WriteBuffer.h>
+#include <IO/WriteBufferValidUTF8.h>
+
+namespace DB
+{
+
+template <typename Base, typename... Args>
+class IIOutputFormatWithUTF8Validation : public Base
```

Comment:
Perhaps this class should be called "IOOutputFormatWithUTF8ValidationAdapter"

Reply:
It's a good idea to call it `Adapter`, but not sure about `IO` prefix, it's will be a bit strange I think, so let's make it `OutputFormatWithUTF8ValidationAdapter ` and ``